packages/block-editor/src/components/block-navigation/index.js

```diff
@@ -67,10 +67,11 @@ export default compose(
 		} = select( 'core/block-editor' );
 		const selectedBlockClientId = getSelectedBlockClientId();
 		return {
-			rootBlocks: getBlocks(),
+			rootBlocks: getBlocks( '', true ),
```

Comment:
Instead of having this flag here, could we instead check if it's a controlled container per item and render a sublist maybe? This could allow us to give them a special treatment too (in terms of design I mean)

I'm asking because I'm not very fan of this API, it's too technical for a public API (sometimes it necessary though).

Reply:
As written it's buggy too since the  cache key is not updated when these inner inner blocks change so the result of getBlocks won't change. Do we need two cache keys. I fear this is going to be more complex for something we don't need that often.

Reply:
That's true, I had not given much thought to the cache key. If the flag is passed, we could just ignore the cache.

If we create a separate selector, we can avoid that. Plus, I think this can just... not have a cache. I don't think this kind of feature needs to be optimized exactly because it shouldn't be used that much.

but it does need to be used...  And I could see other places like the e2e tests wanting to use this instead off the existing getBlolcks

Reply:
@youknowriad do you have further thoughts about this? I definitely think we should do this on the data-store level, rather than for any component which needs it, but I'm not sure which route we should go with. I like the idea of a separate selector as a way to reduce complexity with the getBlocks selector

Reply:
@youknowriad @noahtallen wouldn't a separate function be more expressive and preferable here?

Reply:
I'm happy to split this out to another selector :) @youknowriad I'd like to know what you think about this. I think I agree that it would be more expressive (and possibly easier to maintain)

Reply:
https://github.com/WordPress/gutenberg/pull/24835